### DISCUSSION TITLE
Cannot BUILD in Ubuntu

### DISCUSSION THREAD

**Discussion by pbecchi** [JSBSim-Team/jsbsim#550] [Support requests]
Sorry but I know very little about building in Ubuntu  using make command 
I have modified the code (adding a module FGBldc.cpp) and succesfully compile and build using Visual Studio.
Now I need to do it using Ubuntu to have the modified JSBSim run with PX4 sitl.
I have run following commands on a created build directory:
<cmake ..
make
make install>
From printout I see the new modules being included....
But when I run the code it tell me:
<Propulsion:
In file aircraft/F450/Propulsion.xml: line 3 
 Unknown engine type>
Engine file the definition is like this:
<bldc_engine name="DJI E305">
In windows everything works.
I dont understand.Why this appends only in Ubuntu?
It should execute same way in windows and ubuntu........

**Comment by bcoconni** [comment]
Could you please provide more details about what you are installing and in which order ?
Also could you please tell the command you are running that triggers the error message ?

Thanks.

**Comment by pbecchi** [comment]
All source files are on DLBGengine branch of pbecchi/jsbsim fork. This include revised and cleaned up version of FGBlbg.cpp.
All test cases have also been added to aircraft, engine and scripts directories.
I have built using Visual Studio and succesfully run Test_F450_Launch.xml , results are shown and discussed in #333.
Now i am trying to test the same aircraft DJI_F450 using PX4 sitl
To do this I have cloned the fork on a Ubuntu computer and  build it, this way: 
1 I have created a build directory
2 I have modified src/engine/cmakefile.txt to add  FRBldg.cpp and FRBldg.h lines.
3 I have run `cmake ..`
4 I have run `make`
5 I have run `make install`
6 I have runt the test: `JSBSim --script=script/Test_F450_Launch.xml`
7 The program start but stop with message:
`In file aircraft/F450/Propulsion.xml: line 3
Unknown engine type`
This don't append using windows.

**Comment by pbecchi** [comment]
I have found why........

The windows version was cloned  from @mvacanti  fork modified for the BLDC engine.
When I merged changes to my fork , FGPropulsion.cpp was not modified and the new engine type was not recognised.
I modified FGPropulsion.cpp and now it works!
I keep going......
